### PR TITLE
Creation of Profile Model

### PR DESCRIPTION
**Pull request by Habib-Ahmad**
### What this PR does (required):
- Profile model was created

### Any information needed to test this feature (required):
- First name, last name and birth date are required.

### CODE REVIEW

File: server/models/Profile.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import mongoose from 'mongoose'
+
+const profileSchema = mongoose.Schema({
```
Comment (rajivtitus) [inline server/models/Profile.js:3]: The spacing/indentation seems off, use a code formatter to ensure consistent formatting. Please see below for resources on setting it up:

VS Code: https://www.loom.com/share/bb8aaa6234ff4ce1b48306d16e268486
Atom: https://www.loom.com/share/8025d3c6bd8343e4bee1a0cb61ee1698
Sublime Text: https://www.loom.com/share/12ae38b15a774887924c94b56561861c

File: server/models/Profile.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	profile_photo: {
```
Comment (rajivtitus) [inline server/models/Profile.js]: The standard naming convention for field names is camelCase, please change it accordingly

Ditto throughout!

File: server/models/Profile.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    type: String,
+    required: true,
+  },
+	last_name: {
+    type: String,
+    required: true,
+  },
+	gender: {
+    type: String,
+    enum: ['male', 'female'],
```
Comment (rajivtitus) [inline server/models/Profile.js]:
```suggestion
    enum: ['male', 'female', 'other'],
```
Why don't we consider adding 'other' for individuals who identify as non-binary

File: server/models/Profile.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  },
+	availability: [
+    {
+      startTime: Date,
+      endTime: Date,
+    },
+  ],
+})
+
+module.exports = Profile = mongoose.model("profile", profileSchema);
```
Comment (rajivtitus) [inline server/models/Profile.js]:
```suggestion
module.exports = Profile = mongoose.model("Profile", profileSchema);
```
Nit: The standard naming convention for model names is PascalCase :)

File: server/models/Profile.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	phone_number: {
+    type: String,
+  },
+	address: {
+    type: String,
+  },
+	description: {
+    type: String,
+  },
+	availability: [
```
Comment (rajivtitus) [inline server/models/Profile.js]: This design for availability may not fit the use cases of our project but don't worry as we will be revisiting this at a later date 😄

**Review by rajivtitus — COMMENTED** [review]
Great work by thinking ahead and adding additional fields as necessary 💯 
Just a few comments for you to have a look and address, thanks!

File: server/models/Profile.js (hunk 5 above)
Comment (Habib-Ahmad) [reply server/models/Profile.js]: My thinking was that I would do the necessary formatting in the frontend

File: server/models/Profile.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        type: String,
+        enum: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday']
+      },
+      date: Date,
+      startTime: {
+        type: String,
+        enum: [
+          '1 am',
+          '2 am',
+          '3 am',
```
Comment (moffatethan) [inline server/models/Profile.js:44]: This seems unnecessary, we can just have validation that determines if were inputting valid times into the form.

**Review by moffatethan — CHANGES_REQUESTED** [review]

**Review by moffatethan — APPROVED** [review]
Going to approve, but would have liked to see you have a better availability implementation. We'll get to that more when the time comes.

**Pull request merged**